util-taglib/src/com/liferay/taglib/aui/InputTag.java

```diff
@@ -208,7 +208,9 @@ protected void setAttributes(HttpServletRequest request) {
 		int pos = name.indexOf(StringPool.DOUBLE_DASH);
 
 		if (pos != -1) {
-			name = name.substring(pos + 2, name.length() - 2);
+			if ((name.substring(pos + 2)).endsWith(StringPool.DOUBLE_DASH)) {
+				name = name.substring(pos + 2, name.length() - 2);
```

Comment:
Hey Jose,

I agree with not assuming the double dash at the end, but if there is not one, shouldn't we still do a substring from pos+2 to name.length?